### PR TITLE
openobserve: init at 0.6.0

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

https://github.com/zinclabs/zincobserve

ZincObserve is a cloud native observability platform built specifically for logs, metrics, traces and analytics designed to work at petabyte scale.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dit7ya** [comment]
It has been renamed to openobserve. However I am getting the following error on the latest release - 

cargo-vendor-dir> ln: failed to create symbolic link '/nix/store/r5d9y3pj7x0r416nsq43cv66gznp1arz-cargo-vendor-dir/vrl-0.4.0/mwlqgianlq2dxks73dm7h472v5m4vr6z-vrl-0.4.0': Permission denied
error: builder for '/nix/store/rpv8ihchbk1wcmf89wdh5a5b7dw0xzl4-cargo-vendor-dir.drv' failed with exit code 1;
       last 1 log lines:
       > ln: failed to create symbolic link '/nix/store/r5d9y3pj7x0r416nsq43cv66gznp1arz-cargo-vendor-dir/vrl-0.4.0/mwlqgianlq2dxks73dm7h472v5m4vr6z-vrl-0.4.0': Permission denied
       For full logs, run 'nix log /nix/store/rpv8ihchbk1wcmf89wdh5a5b7dw0xzl4-cargo-vendor-dir.drv'.
error: 1 dependencies of derivation '/nix/store/jks4gv9gylivlk6fkpkrf4xhhxpmx47r-openobserve-0.4.7.drv' failed to build

**Comment by happysalada** [comment]
They have 2 crates with the same version (vrl) one with a git dependency and one without.
The last release of that crate was in may. I would bet that when the versions finally converge we won't have that problem anymore.

File: pkgs/servers/search/zincobserve/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/search/zincobserve/default.nix:93]: Please add a short comment(s) why those tests are disabled

File: pkgs/servers/search/zincobserve/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/search/zincobserve/default.nix:78]:
```suggestion
  buildInputs = [
    bzip2
    oniguruma
    xz
    zlib
    zstd
  ] ++ lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.CoreFoundation
    darwin.apple_sdk.frameworks.Security
  ];
```

File: pkgs/servers/search/zincobserve/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/search/zincobserve/default.nix:56]: Symlinking would not work?

File: pkgs/servers/search/zincobserve/default.nix
Comment (drupol) [inline pkgs/servers/search/zincobserve/default.nix:40]: If you are overriding `configurePhase`, `buildPhase`, `checkPhase`, `installPhase` or any other phase, you should not forget about explicitly running pre and post-phase hooks like [their](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L953) [original](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1008) [definitions](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1037) [do](https://github.com/NixOS/nixpkgs/blob/d71a03ad695407dd482ead32d3eddff50092a1c3/pkgs/stdenv/generic/setup.sh#L1078).

It is generally expected that an appropriate pre-phase hook (e.g. `preBuild`) hook will run at the beginning of a phase (e.g. `buildPhase`) and post-phase hook (e.g. `postBuild`) will run at the end. Hooks are normally ran as a part of a phase so if you override a phase as a whole, you will need to add `runHook hookName` calls (e.g. `runHook preBuild`) manually.

Having phases run pre/post-phase hooks is important because many setup hooks insert their own code into them – omitting a hook might therefore prevent some setup hooks required for proper functionality of a package from running. Additionally, hooks are often inserted by developers into the package expression and by users when overriding a package using `overrideAttrs`. Not running them can thus cause confusion why their code is not executed.

## Examples

### Before

```nix
  installPhase = ''
    your commands
  '';
```

### After

```nix
  installPhase = ''
    runHook preInstall

    your commands

    runHook postInstall
  '';
```

Source: https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/missing-phase-hooks.md

File: pkgs/servers/search/zincobserve/default.nix
Comment (drupol) [inline pkgs/servers/search/zincobserve/default.nix:52]: How about adding a comment on why this patch is needed ?

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by happysalada** [comment]
@dit7ya the latest release has removed the custom vrl dependency that was a problem for the build
https://github.com/openobserve/openobserve/releases/tag/v0.5.3
(I had an issue opened about it and upstream notified me).
When you have a moment to update this, it should be good to go.
If for any reason you can't continue this, I'm happy to take over.

**Comment by dit7ya** [comment]
@happysalada Thanks. I have updated the derivation but yet haven't found time to check if the build succeeds or if the package works.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 232891` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>openobserve</li>
  </ul>
</details>

**Comment by happysalada** [comment]
I just tested the --help, but it's looking good.
I would say it's good enough for us to test and make a module. (or for someone else to start hacking at a module).

**Comment by dit7ya** [comment]
Seems like upstream has releases 0.6.0 and deleted the 0.5.3 release from GitHub. Not cool :(

https://github.com/openobserve/openobserve/releases

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 232891` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>openobserve</li>
  </ul>
</details>

File: pkgs/servers/search/openobserve/default.nix
Comment (happysalada) [inline pkgs/servers/search/openobserve/default.nix:125]: you can remove the todo here.

**Comment by happysalada** [comment]
@drupol I think your comments have been addressed, could you check when you have a moment ? (just to make sure ).

**Comment by happysalada** [comment]
@dit7ya I think this is ready to merge (modulo removing the TODO), what do you think ?

**Review by drupol — APPROVED** [review]
LGTM now but I haven't tested it.

**Comment by happysalada** [comment]
0.6.1 has just been released, maybe we should just wait for a week to see if things calm down a little

**Comment by happysalada** [comment]
ofborg is giving test failures, looks like some missing database setup step
https://logs.ofborg.org/?key=nixos/nixpkgs.232891&attempt_id=3a9ce057-2020-47af-a42f-0e0417de9763

**Comment by happysalada** [comment]
Should we just update to 0.6.4 and call it a day ?

**Comment by happysalada** [comment]
it looks like the version updates have stopped for now.
Would you mind updating to latest ? I think this is good to merge after.

Also I'm happy to be added as a maintainer if you are ok with that.

**Comment by happysalada** [comment]
@dit7ya looking at ofborg it looks like we have some test failures though
```
service::db::compact::file_list::tests::test_file_list_offset
       >     service::db::compact::file_list::tests::test_file_list_process_offset
       >     service::db::compact::files::tests::test_files
       >     service::db::user::tests::test_user
       >     service::triggers::tests::test_triggers
```

**Comment by happysalada** [comment]
closing in favor of https://github.com/NixOS/nixpkgs/pull/270730

**Pull request closed without merging**
